File: 11-container-with-most-water/container-with-most-water.java
class Solution {
public  int maxArea(int[] heights) {
  int maxArea = 0;
  int left = 0, right = heights.length - 1;
  while (left < right) {
    int area = Math.min(heights[left], heights[right]) * (right - left);
    maxArea = Math.max(maxArea, area);
    if (heights[left] < heights[right]) {
      left++;
    } else {
      right--;
    }
  }
  return maxArea;
}
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r = 0,len(height)-1
        best = 0
        while l<r:
            best = max(best,min(height[l],height[r])*(r-l))
            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return best